src/wallet/rpcwallet.cpp

```diff
@@ -3659,8 +3661,27 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
         bool isZaddr = false;
         CTxDestination taddr = DecodeDestination(address);
         if (!IsValidDestination(taddr)) {
-            if (IsValidPaymentAddressString(address)) {
+            auto res = DecodePaymentAddress(address);
+            if (IsValidPaymentAddress(res)) {
                 isZaddr = true;
+
+                bool toSapling = boost::get<libzcash::SaplingPaymentAddress>(&res) != nullptr;
+                noSproutAddrs = noSproutAddrs && toSapling;
+
+                // If we are sending from a shielded address, all recipient
+                // shielded addresses must be of the same type.
+                if (!fromTaddr) {
+                    if (!fromSapling && toSapling) {
```

Comment:
With the suggestion above to add ``fromSprout``, the ``if (!fromTaddr)`` is no longer necessary and this becomes ``if (fromSprout && toSapling)``, which seems clearer.

Reply:
@daira The latest commit addresses this point.